marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -305,6 +302,70 @@ function getPrimaryValue(entityInstance, entityDefinition) {
   }
 }
 
+//Helper funciton to add properties to each result instance under results array in searchReponse
+function buildResult(result, selectedPropertyMetadata, entityInfo, selectedEntityOption) {
```

Comment:
This function seems like it has two chunks of code in it - one for entity-specific properties, and one for generic properties. For that reason, I think this should be two separate functions with names that imply what they do - e.g.

- addEntitySpecificProperties
- addGenericEntityProperties

Reply:
Done